src/common/ipfs.ts

```diff
@@ -67,6 +71,21 @@ export async function uploadMetadata(
   if (typeof metadata === "string") {
     return metadata;
   }
+  async function _fileHandler(object: any){
+    const keys = Object.keys(object)
+    for(const key in keys) {
+      if (Buffer.isBuffer(object[keys[key]])) {
+        object[keys[key]] = await uploadToIPFS(object[keys[key]], contractAddress, signerAddress);
+      }
+      if (typeof object[keys[key]] === "object") {
```

Comment:
is it possible to do `|| typeof File`? @atbe

Reply:
We have a type we can use here called `FileOrBuffer` that should be preferred.